lib/guardian/plug/ensure_not_authenticated.ex

```diff
@@ -0,0 +1,85 @@
+defmodule Guardian.Plug.EnsureNotAuthenticated do
+  @moduledoc """
+  This plug ensures that a invalid JWT was provided and has been
+  verified on the request.
+
+  If one is not found, the `authenticated/2` function is invoked with the
+  `Plug.Conn.t` object and its params.
+
+  ## Example
+
+      # Will call the authenticated/2 function on your handler
+      plug Guardian.Plug.EnsureAuthenticated, handler: SomeModule
+
+      # look in the :secret location.  You can also do simple claim checks:
+      plug Guardian.Plug.EnsureAuthenticated, handler: SomeModule, key: :secret
+
+      plug Guardian.Plug.EnsureAuthenticated, handler: SomeModule, aud: "token"
+
+  If the handler option is not passed, `Guardian.Plug.ErrorHandler` will provide
+  the default behavior.
+  """
```

Comment:
`already_authenticated` sounds like a pretty reasonable thing to call this I think. I don't know what the default would do though. maybe redirect to "/" ? That could possibly cause an infinite redirect though if "/" was EnsureNotAuthenticated. Perhaps a required config option for redirection?